### ISSUE TITLE
Default app template has errors when used with null safety

### ISSUE THREAD

**Issue by mit-mit** [flutter/flutter#67447]
## Repro steps:

1. `flutter create foobar`
1. Add a new file `analysis_options.yaml` containing:
   ```
   analyzer:
     enable-experiment:
       - non-nullable
   ```
1. Edit `pubspec.yaml` to change `sdk: ">=2.7.0 <3.0.0"` to `sdk: ">=2.11.0-0.0 <3.0.0"`
1. Run `flutter analyze --enable-experiment=non-nullable`

**Expected result**: no errors

**Actual result**: Three errors

```
  error • The parameter 'key' can't have a value of 'null' because of its type, and no non-null default value is
         provided • lib/main.dart:31:19 • missing_default_value_for_parameter
  error • The parameter 'title' can't have a value of 'null' because of its type, and no non-null default value is
         provided • lib/main.dart:31:29 • missing_default_value_for_parameter
  error • An expression whose value can be 'null' must be null-checked before it can be dereferenced •
         lib/main.dart:101:22 • unchecked_use_of_nullable_value
```

## SDK version

```
mit-macbookpro4:tmp mit$ flutter --version
Flutter 1.23.0-8.0.pre.167 • channel master • https://github.com/flutter/flutter.git
Framework • revision f551178408 (11 hours ago) • 2020-10-06 11:36:50 +0200
Engine • revision 443cd5a1e1
Tools • Dart 2.11.0 (build 2.11.0-186.0.dev)
```

**Comment by mit-mit** [comment]
cc @Hixie, as discussed in todays meeting

**Comment by Hixie** [comment]
Well you also have to run `dart migrate`. :-)

We will probably want to change this very near the end of the migration to null safety, around the time we port API sample code.

**Comment by TahaTesser** [comment]
```
tahatesser@Tahas-MacBook-Pro master_flutter % flutterm analyze --enable-experiment=non-nullable
Analyzing master_flutter...                                             

  error • The parameter 'key' can't have a value of 'null' because of its type, and no non-null default value is provided •
         lib/main.dart:31:19 • missing_default_value_for_parameter
  error • The parameter 'title' can't have a value of 'null' because of its type, and no non-null default value is provided •
         lib/main.dart:31:29 • missing_default_value_for_parameter
  error • An expression whose value can be 'null' must be null-checked before it can be dereferenced • lib/main.dart:101:22 •
         unchecked_use_of_nullable_value

3 issues found. (ran in 11.5s)
```
<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.23.0-8.0.pre.185, on Mac OS X 10.15.7 19H2
    x86_64, locale en-GB)
    • Flutter version 1.23.0-8.0.pre.185 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision 6620f831e9 (8 hours ago), 2020-10-06 18:15:15 -0700
    • Engine revision 7e6191de07
    • Dart version 2.11.0 (build 2.11.0-186.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.49.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

 
[✓] Connected device (4 available)            
    • iPhone 11 (mobile) • AFF9AE1D-D6C7-46C4-9288-A98DB74D942B • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • Mac OS X 10.15.7 19H2 x86_64
    • Web Server (web)   • web-server                           • web-javascript
      • Flutter Tools
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 85.0.4183.121

• No issues found!
```

</details>

**Comment by Hixie** [comment]
cc @goderbauer

**Comment by goderbauer** [comment]
I just looked at upgrading our app template to NNBD and ran into another issue: The integration test we generate by default imports integration_test package (https://pub.dev/packages/integration_test), which has not been migrated to nnbd yet. This results in the following analyzer warning when upgrading the app template:

```
   info • The library 'package:integration_test/integration_test.dart' is legacy, and should not be imported into a null safe library • integration_test/app_test.dart:10:8 •
          import_of_legacy_library_into_null_safe
   info • The library 'package:integration_test/integration_test_driver.dart' is legacy, and should not be imported into a null safe library • integration_test/driver.dart:6:8 •
          import_of_legacy_library_into_null_safe
```

Are there plans for migrating that package soon or should we remove the integration test from the app template for the NNBD launch? (/cc @dnfield)

**Comment by dnfield** [comment]
This was blocked by vm_sercice but should be ready to migrate now

**Comment by goderbauer** [comment]
vm_service doesn't have a null-safe version either, though?

**Comment by dnfield** [comment]
It was recently updated.

**Comment by goderbauer** [comment]
NNBD mgiration of vm_service is still in progress: https://dart-review.googlesource.com/c/sdk/+/156980

**Comment by Hixie** [comment]
To be clear the goal here is to support `flutter create; dart migrate` without errors here; we aren't planning on migrating the template itself until after the next stable release (i.e. probably in Q2 sometime).

**Comment by goderbauer** [comment]
This is blocked by https://github.com/flutter/flutter/issues/71379.

**Comment by pcsosinski** [comment]
@goderbauer i believe 71379 is no longer blocking this?

**Comment by dnfield** [comment]
We've backed integration_test out of the template, this should be unblocked.

**Comment by mit-mit** [comment]
Thinking more about this, I'm wondering if we should push this out by a release to ensure we have enough of an ecosystem of null safety packages in place before we default new apps/packages to it.

**Comment by goderbauer** [comment]
@pcsosinski This is no longer blocked, correct. And I've posted https://github.com/flutter/flutter/pull/74090 to add a test to ensure that the templates can be migrated to NNBD.

**Comment by goderbauer** [comment]
@mit-mit 

The goal of this bug as I understand it is to make sure that the default app template can be migrated to NNBD with `dart migrate`, not to turn NNBD on by default.

**Comment by mit-mit** [comment]
Discussed in release meeting, where we came to this suggestion:

* We'll not switch the template to null safety by default in the current release. We'll try to do that in the following stable release.
* We'll add a bit of output towards the end of `flutter create` that explains how to enable null safety:

```
All done!

In order to run your application, type:

  $ cd foobar40
  $ flutter run

To enable null safety, type:
  $ dart migrate --apply-changes

Your application code is in foobar40/lib/main.dart.
```

@Hixie please review this message and confirm that is what you had in mind.

**Comment by flutterq** [comment]
All possible solution added [[Solved] The parameter ‘key’ can’t have a value of ‘null’ because of its type, and no non-null default value is provided](https://fluttercorner.com/the-parameter-key-cant-have-a-value-of-null-because-of-its-type-and-no-non-null-default/)
